File: CPP/Motherboard.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Ram.cpp"
#include "Harddrive.cpp"
#include "Ssd.cpp"
#include "Vga.cpp"
#include "Psu.cpp"
#include "Casing.cpp"

// Kelas Motherboard yang merepresentasikan sebuah motherboard dengan berbagai komponen
class Motherboard {
private:
    Cpu cpu;
    vector<Ram> ramList;
    Harddrive harddrive;  
    Ssd ssd;
    Vga vga;
    Psu psu;
    Casing casing;

public:
    Motherboard() {}


    // Konstruktor dengan parameter untuk menginisialisasi semua komponen
    Motherboard(Cpu cpu, vector<Ram> ramList, Harddrive harddrive, Ssd ssd, Vga vga, Psu psu, Casing casing) {
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddrive = harddrive;  
        this->ssd = ssd;
        this->vga = vga;
        this->psu = psu;
        this->casing = casing;
    }

    //Setter
    void setCpu(Cpu cpu) {
        this->cpu = cpu;
    }

    void setRamList(vector<Ram> ramList) {
        this->ramList = ramList;
    }

    void addRam(Ram ram) {
        this->ramList.push_back(ram);
    }

    void setHarddrive(Harddrive harddrive) {  
        this->harddrive = harddrive;
    }

    void setSsd(Ssd ssd) {
        this->ssd = ssd;
    }

    void setVga(Vga vga) {
        this->vga = vga;
    }

    void setPsu(Psu psu) {
        this->psu = psu;
    }

    void setCasing(Casing casing) {
        this->casing = casing;
    }


    //Getter
    Cpu getCpu() {
        return cpu;
    }

    vector<Ram> getRamList() {
        return ramList;
    }

    Harddrive getHarddrive() {  
        return harddrive;
    }

    Ssd getSsd() {
        return ssd;
    }

    Vga getVga() {
        return vga;
    }

    Psu getPsu() {
        return psu;
    }

    Casing getCasing() {
        return casing;
    }

    ~Motherboard() 
    {
        
    }
};

File: python/Motherboard.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive
from Ssd import Ssd
from Vga import Vga
from Psu import Psu
from Casing import Casing

class Motherboard:
    #atribut private
    __cpu = ""
    __ram_list = []
    __harddrive = ""
    __ssd = ""
    __vga = ""
    __psu = ""
    __casing = ""

    #constructor
    def __init__(self, cpu, ram_list, harddrive, ssd, vga, psu, casing):
        self._cpu = cpu
        self._ram_list = ram_list
        self._harddrive = harddrive
        self._ssd = ssd
        self._vga = vga
        self._psu = psu
        self._casing = casing

    #setter
    def set_cpu(self, cpu):
        self._cpu = cpu

    def set_ram_list(self, ram_list):
        self._ram_list = ram_list

    def add_ram(self, ram):
        self._ram_list.append(ram)

    def set_harddrive(self, harddrive):
        self._harddrive = harddrive

    def set_ssd(self, ssd):
        self._ssd = ssd

    def set_vga(self, vga):
        self._vga = vga

    def set_psu(self, psu):
        self._psu = psu

    def set_casing(self, casing):
        self._casing = casing


    #getter
    def get_cpu(self):
        return self._cpu

    def get_ram_list(self):
        return self._ram_list

    def get_harddrive(self):
        return self._harddrive

    def get_ssd(self):
        return self._ssd

    def get_vga(self):
        return self._vga

    def get_psu(self):
        return self._psu

    def get_casing(self):
        return self._casing